src/mod_inbox_muclight.erl

```diff
@@ -0,0 +1,164 @@
+    _ ->
+      write_to_inbox(Server, User, UserFromRoom, Sender, MsgId, Packet)
+  end.
+
+-spec handle_system_message(User :: jid:jid(),
+                            Server :: host(),
+                            UserFromRoom :: jid:jid(),
+                            Packet :: exml:element(),
+                            MsgId :: id(),
+                            WriteAffChanges :: boolean()) -> ok.
+handle_system_message(_User, _Server, _UserFromRoom, _Packet, _MsgId, false) ->
+  ok;
+handle_system_message(User, Server, UserFromRoom, Packet, MsgId, true) ->
+  case {check_invitation_message(User, Packet),
+        check_new_owner_message(User, Packet)} of
+    {false, false} ->
+      case check_kicked_message(User, Packet) of
+        true ->
+          handle_kicked_message(User, Server, UserFromRoom);
+        false ->
+          %% some other system muc light message. Do not write to inbox.
+          ok
+      end;
+    _ ->
+      handle_invitation_message(User, Server, UserFromRoom, Packet, MsgId)
+  end.
+
+-spec handle_invitation_message(User :: jid:jid(),
+                                Server :: host(),
+                                UserFromRoom ::  jid:jid(),
+                                Packet ::  exml:element(),
+                                MsgId ::  id()) -> ok.
+handle_invitation_message(User, Server, UserFromRoom, Packet, MsgId) ->
+  BareRemJID = jid:to_bare(UserFromRoom),
+  mod_inbox_utils:write_to_receiver_inbox(Server, UserFromRoom, User, BareRemJID, MsgId, Packet).
+
+-spec handle_kicked_message(User :: jid:jid(),
+                            Server :: host(),
+                            UserFromRoom ::  jid:jid()) -> ok.
+handle_kicked_message(User, Server, UserFromRoom) ->
+  CheckRemove = mod_inbox_utils:check_remove_on_kicked(Server),
+  maybe_remove_inbox_row(User, Server, UserFromRoom, CheckRemove).
+
+-spec maybe_remove_inbox_row(Username :: jid:jid(),
+                             Server :: host(),
+                             RemoteJid :: jid:jid(),
+                             CheckRemote :: boolean()) -> ok.
+maybe_remove_inbox_row(_, _, _ , false) ->
+  ok;
+maybe_remove_inbox_row(Username, Server, RemoteJid, true) ->
+  UBin = jid:to_binary(jid:to_bare(Username)),
+  RemoteBin = jid:to_binary(RemoteJid),
+  mod_inbox_backend:remove_inbox(UBin, Server, RemoteBin).
+
+-spec write_to_inbox(Server :: host(),
+                     RemoteIsSender :: jid:jid(),
+                     User :: jid:jid(),
+                     RemoteIsSender :: jid:jid(),
+                     MsgId :: id(),
+                     Packet :: exml:packet()) -> ok.
+write_to_inbox(Server, RemoteIsSender, User, RemoteIsSender, MsgId, Packet) ->
+  BareRemJID = jid:to_bare(RemoteIsSender),
+  mod_inbox_utils:write_to_sender_inbox(Server, RemoteIsSender, User, BareRemJID, MsgId, Packet);
+write_to_inbox(Server, Remote, User, Sender, MsgId, Packet) ->
```

Comment:
This function is called like this: `write_to_inbox(Server, User, UserFromRoom, Sender, MsgId, Packet)`, so `Remote` and `User` are probably swapped?